src/cpp/main.cpp

```diff
@@ -0,0 +1,69 @@
+
+#define PDE_OUT_VERBOSE
+#include "utilities/mesh_io.hpp"
+#include "utilities/visualize.hpp"
+#include "solver/complete_solver.hpp"
+
+#include <ctime>
+#include "solver/derivatives.hpp"
+
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+#include <fenv.h>
+#include <string>
+
+#include <utility>
+
+void
+parse_params(int argc, char* argv[], std::vector<double>& values) {
+	for (int i = 1; i < argc; ++i) {
+		values[i-1] = std::stod(std::string(argv[i]));
+	}
+}
+
+using namespace dealii;
+
+int main(int argc, char* argv[]) {
+
+	// Parameters for the model are taken in input from the line command
+	// example activation: 
+	//  mpirun -n 4 ./main 2 0.1 0.04 0.3 4.5 4.19 6.5 2.5 10.4 0.130 15.255 2.0 0.5 9.0
+
+	Utilities::MPI::MPI_InitFinalize mpi_init(argc, argv);
+
+	// ENABLE ALL EXCEPTIONS FOR NAN GENERATION TO SEE WHERE THINGS GO ROGUE
+	feenableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
+
+	std::vector<double> vals(100);
+	parse_params(argc, argv, vals);
```

Comment:
You should either look into `dealii::ParameterHandler` and the associated classes, or a better parser for command line arguments (such as [clipp](https://github.com/muellan/clipp)), or both. This approach is extremely error prone - I imagine it's not too hard for a user to swap two command line arguments by mistake.